src/Razor/src/Microsoft.CodeAnalysis.Razor.CohostingShared/DocumentSymbol/CohostDocumentSymbolEndpoint.cs

```diff
@@ -50,7 +51,16 @@ public ImmutableArray<Registration> GetRegistrations(VSInternalClientCapabilitie
         => request.TextDocument.ToRazorTextDocumentIdentifier();
 
     protected override Task<SumType<DocumentSymbol[], SymbolInformation[]>?> HandleRequestAsync(DocumentSymbolParams request, RazorCohostRequestContext context, CancellationToken cancellationToken)
-        => HandleRequestAsync(context.TextDocument.AssumeNotNull(), _useHierarchicalSymbols, cancellationToken);
+    {
+        // The editor can send us a document symbol request in a .NET Framework project for some reason, so we have to be
+        // a little defensive here.
+        if (context.TextDocument is null)
```

Comment:
I was tempted to put this null check in Roslyn, or a base class in this repo, but it seems like there are only two endpoints that need it (for .NET Framework projects at least) so I'll leave that for https://github.com/dotnet/razor/issues/11942 when I tackle incompatible .NET Core projects